### PR TITLE
 Clarified documentation for `next` in LazyList

### PR DESCRIPTION
**Pull request by mars0i**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
So, seven commits just to change the ocamldoc for LazyList.next?

**Comment by mars0i** [comment]
I had to undo some junk that was there because I had originally worked on master for a different PR.  I didn't want to just blow away my fork and start over, because I have another PR outstanding (which overwrote the junk).  My mess.  Sorry about that.

File: src/batLazyList.mli [hunk 1]
```diff
@@ -287,7 +287,9 @@ val rindex_ofq : 'a -> 'a t -> int option
 *)
 
 val next : 'a t -> 'a node_t
-(**Compute and return the next value of the list*)
+(** Compute and return the first node from the list as a [Cons].  This 
+    differs from [hd], which returns first element (the first component of
```
Comment (gasche) [inline src/batLazyList.mli]: which returns *the* first element

**Comment by gasche** [comment]
Could you rebase your branch to remove the extra commits? One simple way to do it would be the following:

```
git checkout -b tmp # create a temporary branch in which to work
git cherry-pick 6b619cd # the hash of the commit actually doing the work in this branch
git checkout lazy_next_doc
git reset --hard tmp
git branch --delete tmp

**Comment by mars0i** [comment]
Thanks very much for the git wizardry, @gasche.  

Did that do it?  Still looks a little noisy at my end.

**Comment by mars0i** [comment]
Github now allows "squashing" multiple commits into a single commit.  It looks like this has to done in the main repo, or enabled there, at least  https://github.com/blog/2141-squash-your-commits

**Comment by gasche** [comment]
I will squash indeed as this is the least work-intensive way to resolve this PR, but I am a bit worried by the fact that, out of three different pull-requests, you were unable to offer a proper commit history. This is a useful skill to acquire, and it would be necessary for larger contributions where preserving the commit history is important (it contains several independent changes) -- which I hope you will have the occasion to contribute to Batteries. For this reason, my (implicit) offer to try to find out what's the issue and debug the commit history still stands -- or we may have this discussion again on your next PR.

**Comment by mars0i** [comment]
Thanks @gasche.  

(I admit I am a bit puzzled about why git makes it so tricky to clean up a commit history.  My work habit is to make many little commits as I go along.  This is a safe strategy.  The problem is that no matter what branch this is done on, and what branch it is copied too, by default git wants to preserve every little bit of commit history.  I'm beginning to think that the best thing might be to deceive git by making a second repo on my machine that bears no relationship to the main parent repo, and then, when the code seems to be settled, copy it into a single commit in my main local copy of  the repo.  Or something like that.)

**Comment by gasche** [comment]
In theory `rebase` is designed to help you transfer commits across different histories, see its documentation `git rebase --help` for more details.

**Comment by mars0i** [comment]
@gasche, I'll study `rebase`--something I've been aware of but have avoided until now.

**Pull request merged**